### ISSUE TITLE
GridSchemaV99 possibly misconfigured. Results in a JsonSyntaxException when using setNfolds() on algorithm.

### ISSUE THREAD

**Issue by kmullapudi-11** [h2oai/sparkling-water#2176]
Providing us with the observed and expected behavior definitely helps. Giving us with the following information definitively helps:

- Sparkling Water/PySparkling/RSparkling version
- Hadoop Version & Distribution
- Execution mode `YARN-client`, `YARN-cluster`, standalone, local ..
- YARN logs in case of running on yarn. To collect such a logs you may run `yarn logs -applicationId <application ID>` where the application ID is displayed when Sparkling Water is started
- H2O & Spark logs if not running on YARN. You can find these logs in Spark work directory
- Are you using Windows/Linux/MAC?
- Spark & Sparkling Water configuration including the memory configuration

**Environment Details**
I'm using using the following Maven package for Sparkling Water: `ai.h2o:sparkling-water-package_2.11:3.30.0.4-1-2.4`. I'm running the code locally in a Zeppelin notebook inside a docker container. I'm using a Mac and using Spark version 2.4.4, Scala Version 2.11.12. However, I was able to reproduce this issue even by changing tests to the below mentioned code within the `H2OGridSearchTestSuite.scala` file.

**Issue**
When I `setNfolds()` for any of the `H2OSupervisedAlgorithms`, do grid search using `H2OGridSearch` and then call `fit()` on the dataset, I get a `JsonSyntaxException`. I was able to get the expected data by setting a breakpoint inside the `RestCommunication.scala` file. The error seems to be thrown by a misconfigured `GridSchemaV99` which results in an error when `deserialize()` is called the `RestCommunication.request` method Line 169. 

```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 608096 path $.cross_validation_metrics_summary[0].data[0][0]
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:224)
  at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
  at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
  at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
  at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129)
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220)
  at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
  at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129)
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220)
  at com.google.gson.Gson.fromJson(Gson.java:887)
  at com.google.gson.Gson.fromJson(Gson.java:852)
  at com.google.gson.Gson.fromJson(Gson.java:801)
  at ai.h2o.sparkling.backend.utils.RestCommunication$class.ai$h2o$sparkling$backend$utils$RestCommunication$$deserialize(RestCommunication.scala:164)
  at ai.h2o.sparkling.backend.utils.RestCommunication$$anonfun$request$1.apply(RestCommunication.scala:147)
  at ai.h2o.sparkling.backend.utils.RestCommunication$$anonfun$request$1.apply(RestCommunication.scala:145)
  at ai.h2o.sparkling.utils.ScalaUtils$.withResource(ScalaUtils.scala:28)
  at ai.h2o.sparkling.backend.utils.RestCommunication$class.request(RestCommunication.scala:145)
  at ai.h2o.sparkling.ml.algos.H2OGridSearch.request(H2OGridSearch.scala:46)
  at ai.h2o.sparkling.backend.utils.RestCommunication$class.query(RestCommunication.scala:54)
  at ai.h2o.sparkling.ml.algos.H2OGridSearch.query(H2OGridSearch.scala:46)
  at ai.h2o.sparkling.ml.algos.H2OGridSearch.getGridModels(H2OGridSearch.scala:129)
  at ai.h2o.sparkling.ml.algos.H2OGridSearch.fit(H2OGridSearch.scala:163)
  at ai.h2o.sparkling.ml.algos.H2OGridSearch.fit(H2OGridSearch.scala:46)
  at org.apache.spark.ml.Pipeline$$anonfun$fit$2.apply(Pipeline.scala:153)
  at org.apache.spark.ml.Pipeline$$anonfun$fit$2.apply(Pipeline.scala:149)
  at scala.collection.Iterator$class.foreach(Iterator.scala:891)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
  at scala.collection.IterableViewLike$Transformed$class.foreach(IterableViewLike.scala:44)
  at scala.collection.SeqViewLike$AbstractTransformed.foreach(SeqViewLike.scala:37)
  at org.apache.spark.ml.Pipeline.fit(Pipeline.scala:149)
  ... 59 elided
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 608096 path $.cross_validation_metrics_summary[0].data[0][0]
  at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:385)
  at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:213)
  ... 90 more
```

Please also provide us with the full and minimal reproducible code.
```
import org.apache.spark.h2o._
import ai.h2o.sparkling.ml.algos._
import ai.h2o.sparkling.ml.models._
import org.apache.spark.ml.Pipeline

val hc = H2OContext.getOrCreate()
import hc._
import hc.implicits._

var data = spark.read.format("csv").option("header", "true").option("inferSchema", "true").load("/path/to/prostate.csv")
// Link to data: https://github.com/h2oai/sparkling-water/blob/master/examples/smalldata/prostate/prostate.csv

val drf =  new H2ODRF()
    .setFeaturesCols(Array("AGE", "RACE", "DPROS", "DCAPS", "PSA", "VOL", "GLEASON"))
    .setLabelCol("CAPSULE")
    .setColumnsToCategorical(Array("RACE", "DPROS", "DCAPS","GLEASON", "CAPSULE"))
    .setSplitRatio(0.8)
    .setNfolds(4)

val nps = Map(
        "ntrees" -> Array(10, 50).map(_.asInstanceOf[AnyRef]))

val search = new H2OGridSearch()
    .setHyperParameters(hyperParams)
    .setAlgo(drf)
    .setStrategy("Random Discrete")

val model = search.fit(data)
```

**Comment by jakubhava** [comment]
@kmullapudi-11 thank you for the great report!

You have discovered a bug which is going to be fixed in the next release 3.30.0.5 https://0xdata.atlassian.net/browse/SW-2312

I will close this issue for now, please use the jira to track the issue

**Comment by kmullapudi-11** [comment]
@jakubhava Thanks for the quick reply and fix! When can I expect the 3.30.0.5 release?

**Comment by jakubhava** [comment]
It should be out around end of June, probably earlier
